modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -260,17 +260,32 @@ public JSONArray getJSONArray(JSONObject json, String tag, JSONArray fallback) {
     /**
      * Access a literal value (string, numeric, or boolean).
      * 
-     * @param index
+     * @param json JSONArray object to parse.
+     * @param index position in the provided array for which to retrieve a value.
      * @return required string, numeric or boolean
      * @throws MBFormatException if required index not available.
      */
     public Object value(JSONArray json, int index) {
+        return value(json, index, false);
+    }
+
+    /**
+     * Access a literal value (string, numeric, boolean or null).
+     *
+     * @param json JSONArray object to parse.
+     * @param index position in the provided array for which to retrieve a value.
+     * @param treatNullAsLiteral true if nulls should be treated as literals, false otherwise.
+     * @return required string, numeric, boolean or null.
+     * @throws MBFormatException if required index not available.
+     */
+    public Object value(JSONArray json, int index, boolean treatNullAsLiteral) {
```

Comment:
I'm not sure how this will affect the MBFilter class.  I'd want to double check this to make sure filters are still working as expected.

Reply:
This method is new, and has the `treatNullAsLiteral` flag. The method that existed prior to this PR didn't and should call this one with that flag set to false, which should have the same behavior as before. However, since I refactored the Decision logic into functions, I may not need it any more.

Reply:
... and I don't, so I'll revert this back to the way it was.